optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/ConstraintSessionFactory.java

```diff
@@ -23,7 +23,7 @@
     /**
      * This method is thread-safe.
      * @param constraintMatchEnabled  true if {@link ScoreDirector#isConstraintMatchEnabled()} should be true
-     * @param workingSolution never null
+     * @param workingSolution if null, uniform synthetic constraint weights may be applied
```

Comment:
"may" or "will"?

Reply:
to take that philosophy a step further: if it's indeed "may", then it should specify when and when it doesn't happen, if that's practical.